### PR TITLE
Refactor bad option value stash in prevision of handling multiple type of messages

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Refactor prior to Refs #6824

### CODE REVIEW

File: pylint/config/callback_actions.py [hunk 1]
```diff
@@ -365,7 +365,36 @@ def __call__(
...
 
-class _DisableAction(_AccessLinterObjectAction):
+class _XableAction(_AccessLinterObjectAction):
+    """Callback action for enabling or disabling a message."""
+
+    def _call(
+        self,
+        xabling_function: Callable[[str], None],
+        values: str | Sequence[Any] | None,
+        option_string: str | None = "--disable",
```
Comment (DanielNoord) [inline pylint/config/callback_actions.py:375]:
```suggestion
        option_string: str | None,
```

**Review by DanielNoord — APPROVED** [review]
One nit!

Thanks for splitting up 👍

File: pylint/config/callback_actions.py [hunk 2]
```diff
@@ -365,7 +365,36 @@ def __call__(
...
+                    (option_string, msgid)
+                )
+
+    @abc.abstractmethod
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: str | Sequence[Any] | None,
+        option_string: str | None = "--disable",
```
Comment (Pierre-Sassoulas) [inline pylint/config/callback_actions.py:392]:
```suggestion
        option_string: str | None,
```

**Comment by Pierre-Sassoulas** [comment]
> Thanks for splitting up 

It was already separated by commit in the other pull request so it was easy 😄

**Pull request merged**